src/coreclr/jit/lower.cpp

```diff
@@ -2249,6 +2230,297 @@ GenTree* Lowering::LowerTailCallViaJitHelper(GenTreeCall* call, GenTree* callTar
     return result;
 }
 
+//------------------------------------------------------------------------
+// LowerCFGCall: Potentially lower a call to use control-flow guard. This
+// expands indirect calls into either a validate+call sequence or to a dispatch
+// helper taking the original target in a special register.
+//
+// Arguments:
+//    call         -  The call node
+//
+void Lowering::LowerCFGCall(GenTreeCall* call)
+{
+    if (call->IsHelperCall(comp, CORINFO_HELP_VALIDATE_INDIRECT_CALL) ||
+        call->IsHelperCall(comp, CORINFO_HELP_DISPATCH_INDIRECT_CALL))
+    {
+        return;
+    }
```

Comment:
Does this happen? i.e., would we expect to see these helper calls come through this path?

Reply:
It happens for the validator but not the dispatcher -- changed the second disjunct to be in an assert instead.
FWIW, we do see indirect calls to the dispatcher/validator that go through most of the normal paths for calls, so there are a few special cases to handle this as they are generally the only non-checked indirect calls we should see.